src/app/course-page/components/course-page/course-page.component.ts

```diff
@@ -1,62 +1,136 @@
 import { Component, OnInit, SimpleChanges, OnDestroy, ChangeDetectionStrategy } from '@angular/core';
 import { CourseItem } from '../../models/course-item.model';
 import { ItemCourseService } from '../../services/item-course.service';
-import { Subscription } from 'rxjs';
-import { Router } from '@angular/router';
+import { ShowParamsService } from '../../services/show-params.service';
+import { AuthorizationService } from  '../../../shared/services/authorization.service';
+import { LoadingService }  from '../../../shared/services/loading.service';
+import { SubscriptionLike, Subject } from 'rxjs';
+import { Router, ActivatedRoute, Params } from '@angular/router';
+import { debounceTime, distinctUntilChanged } from 'rxjs/operators';
 
 @Component({
   selector: 'app-course-page',
   templateUrl: './course-page.component.html',
   styleUrls: ['./course-page.component.scss'],
-  changeDetection: ChangeDetectionStrategy.OnPush
+  // changeDetection: ChangeDetectionStrategy.OnPush
 })
 export class CoursePageComponent implements OnInit, OnDestroy {
 
   public inputValue = '';
-
-  courseItems: CourseItem[];
+  message: string;
+  error: '';
+  courseItems: CourseItem[] = [];
+  private searchTextChanged = new Subject<string>();
 
   public counter: number = 0;
-  public isShowCourse: boolean = false;
-
-  itemsSub: Subscription;
+  subscriptions: SubscriptionLike[] = [];
 
   constructor(
     private itemCourseService: ItemCourseService,
-    private router: Router
+    private showParamsService: ShowParamsService,
+    private router: Router,
+    private route: ActivatedRoute,
+    private auth: AuthorizationService,
+    private loadService: LoadingService
   ) { }
 
   public change(): void {
     this.counter = this.counter + 1;
     console.log(this.counter);
-    // this.isShowCourse = this.isShowCourse ? false : true;
   }
 
-  public onItemDelete(item: CourseItem): void {
-    this.itemCourseService.deleteItem(item);
+  public onItemDelete(item: CourseItem) {
+    // this.itemCourseService.deleteItem(item);
+    this.subscriptions.push(this.itemCourseService.removeItem(item)
+    .subscribe( () => {
+      this.courseItems = this.courseItems.filter( t => t.id !==  item.id);
```

Comment:
do you need this code? After the deletion you call fetchItems and the updated list of items will be shown

Reply:
removed this part of code